### PR TITLE
feat(table): add `filterOnClose` prop for `Column`

### LINKED ISSUE #43908: When clicked outside the table filters drop-down, it also apply the changes
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-gwdjtd)

### Steps to reproduce

1. Click on the filters icon on Name column
2. Select "Joe"
3. Click on any area outside the dropdown
4. Observe the table data

### What is expected?

The table filters should be applied when clicked on Okay/Reset button but they are applying with outside clicks as well. I want a way to disable this

### What is actually happening?

The table filters are applying with outside clicks as well.

| Environment | Info |
| --- | --- |
| antd | 4.24.1 |
| React | 18.2.0 |
| System | Ubuntu 20.04.6 LTS |
| Browser | Google Chrome 115.0.5790.102 |

### PR DESCRIPTION
**Pull request by xsjcTony**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #43908

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add an option to not trigger the filter when the dropdown menu closes |
| 🇨🇳 Chinese | 添加在下拉菜单关闭时不触发筛选的选项 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

File: components/table/index.en-US.md [hunk 1]
```diff
@@ -192,6 +192,7 @@ One of the Table `columns` prop for describing the table's columns, Column has t
 | filtered | Whether the `dataSource` is filtered | boolean | false |  |
 | filteredValue | Controlled filtered value, filter icon will highlight | string\[] | - |  |
 | filterIcon | Customized filter icon | ReactNode \| (filtered: boolean) => ReactNode | - |  |
+| filterOnClose | Whether to trigger filter when the filter menu closes | boolean | true |  |
```
Comment (li-jia-nan) [inline components/table/index.en-US.md:195]: 版本号没写

File: components/table/index.zh-CN.md [hunk 2]
```diff
@@ -193,6 +193,7 @@ const columns = [
 | filtered | 标识数据是否经过过滤，筛选图标会高亮 | boolean | false |  |
 | filteredValue | 筛选的受控属性，外界可用此控制列的筛选状态，值为已筛选的 value 数组 | string\[] | - |  |
 | filterIcon | 自定义 filter 图标。 | ReactNode \| (filtered: boolean) => ReactNode | false |  |
+| filterOnClose | 是否在筛选菜单关闭时触发筛选 | boolean | true |  |
```
Comment (li-jia-nan) [inline components/table/index.zh-CN.md:196]: 同上

File: components/table/index.en-US.md (hunk 1 above)
Comment (li-jia-nan) [reply components/table/index.en-US.md:195]: - #47472

File: components/table/index.zh-CN.md (hunk 2 above)
Comment (li-jia-nan) [reply components/table/index.zh-CN.md:196]: - #47472

File: components/table/index.en-US.md (hunk 1 above)
Comment (xsjcTony) [reply components/table/index.en-US.md:195]: 阿巴阿巴🤦‍♀️

**Pull request merged**